app/controllers/uploader/files_controller.rb

```diff
@@ -1,169 +1,169 @@
-      render :index
-    end
-
-    def edit
-      raise "403" unless @cur_node.allowed?(:edit, @cur_user, site: @cur_site)
-      render :edit
-    end
-
-    def show
-      raise "403" unless @cur_node.allowed?(:read, @cur_user, site: @cur_site)
-      render :show
-    end
-
-    def delete
-      raise "403" unless @cur_node.allowed?(:delete, @cur_user, site: @cur_site)
-      render :delete
-    end
-
-    def new_files
-      raise "403" unless @cur_node.allowed?(:edit, @cur_user, site: @cur_site)
-      render :new_files
-    end
-
-    def new_directory
-      raise "403" unless @cur_node.allowed?(:edit, @cur_user, site: @cur_site)
-      render :new_directory
-    end
-
-    def create
-      set_params(:directory, :files)
-      if @directory
-        create_directory
-      elsif @files
-        create_files
-      else
-        raise "400"
-      end
-    end
-
-    def update
-      set_params(:filename, :files, :text, :file)
-      raise "403" unless @cur_node.allowed?(:edit, @cur_user, site: @cur_site)
-      raise "400" unless @filename
-
-      if !@item.directory?
-        @text ? (@item.text = @text) : @item.read
-      end
-      @item.filename = @filename if @filename && @filename =~ /^#{@cur_node.filename}/
-      @item.site = @cur_site
-      result = @item.save
-      @item.path = @item.saved_path unless result
-
-      file = @files.find(&:present?) rescue nil
-      file = @file if @file.present?
-      Fs.binwrite @item.saved_path, file.read if result && file
-
-      location = "#{uploader_files_path}/#{@item.filename}?do=edit"
-      render_update result, location: location
-    end
-
-    def destroy
-      raise "403" unless @cur_node.allowed?(:edit, @cur_user, site: @cur_site)
-
-      if params[:ids].present?
-        destroy_all
-      else
-        render_destroy @item.destroy, location: "#{uploader_files_path}/#{@item.dirname}"
-      end
-    end
-
-    def destroy_all
-      @paths = params[:ids]
-      @paths.each do |path|
-        item = @model.file("#{@item.path}/#{path}")
-        item.destroy
-      end
-      render_destroy true, location: "#{uploader_files_path}/#{@item.filename}"
-    end
-end
+class Uploader::FilesController < ApplicationController
+  include Cms::BaseFilter
+  include Cms::CrudFilter
+
+  model Uploader::File
+
+  before_action :create_folder
+  before_action :redirect_from_index
+  before_action :set_item
+
+  navi_view "uploader/main/navi"
+
+  private
+  def create_folder
+    return if @model.file(@cur_node.path)
+    cur_folder = @model.new path: @cur_node.path, is_dir: true
+    cur_folder.site = @cur_site
+    raise "404" unless cur_folder.save
+  end
+
+  def redirect_from_index
+    return unless params[:filename].blank?
+    redirect_to action: :file, filename: @cur_node.filename
+  end
+
+  def set_item
+    filename = ::CGI.unescape params[:filename]
+    raise "404" if filename != @cur_node.filename && filename !~ /^#{@cur_node.filename}\//
```

Comment:
__[brakeman]__
* Model attribute used in regex [:ref](http://brakemanscanner.org/docs/warning_types/denial_of_service/)

<!-- Comment by [SideCI](https://sideci.com) -->